Google.PowerShell/Sql/GcSqlInstanceCmdlets.cs

```diff
@@ -536,13 +588,90 @@ protected override void ProcessRecord()
             InstancesResource.ImportRequest request = Service.Instances.Import(body, Project, Instance);
             Operation result = request.Execute();
             result = WaitForSqlOperation(result);
-            if (result.Error != null) {
+            if (UploadLocalFile)
+            {
+                bucketService.Objects.Delete(bucketName, fileName).Execute();
+                bucketService.Buckets.Delete(bucketName).Execute();
+            }
+            if (result.Error != null)
+            {
                 foreach (OperationError error in result.Error.Errors)
                 {
+
                     throw new GoogleApiException("Google Cloud SQL API", error.Message + error.Code);
                 }
             }
         }
+
+        private string UploadFile()
+        {
+            Random rnd = new Random();
+            int bucketRnd = rnd.Next(1000000);
+            bucketName = "import" + bucketRnd.ToString();
+            fileName = "toImport";
+            bucketService = new StorageService(GetBaseClientServiceInitializer());
+            CreateBucket();
+            try
+            {
+                UploadObject();
+            }
+            catch (Exception e)
+            {
+                bucketService.Buckets.Delete(bucketName).Execute();
+                throw e;
+            }
+            UpdatePermissions();
```

Comment:
You can probably delete this.

You can set a "defaultAcl" in the bucket or object creation request. And setting the default object Acls on the temp bucket is probably sufficient. e.g.

````
Bucket bucket = new Google.Apis.Storage.v1.Data.Bucket();
bucket.Name = bucketName;

BucketResource.InsertRequest bucketInsertReq = bucketService.Buckets.Insert(bucket, Project);
// The "privateProject" ACL means ..., for more information see ...
bucketInsertReq.DefaultAcl = "projectPrivate";
bucketInsertReq.DefaultObjectAcl = "projectPrivate";

return bucketInsertReq.Execute();